### PR TITLE
Re-instating ids endpoint

### PR DESCRIPTION
**Pull request by peteclark-ft**
The `/__ids` endpoint was removed during the refactoring work for the carousel, because the original did not work as expected when the collection was large. 

The new /__ids endpoint will work the same way, except it will now timeout if the collection has too many records (timeout is set to 10s). Additionally, if the mongo query errors during the opening of the mongo session, or during the iterator creation, we will return a non-200 code.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,11 +8,11 @@ The same data can then be read from here just like from the original CMS.__
 
 You need [Go to be installed](https://golang.org/doc/install). Please read about Go and about [How to Write Go Code](https://golang.org/doc/code.html) before jumping right in. For example you will need Git, Mercurial, Bazaar installed and working, so that Go can use them to retrieve dependencies. For this additionally you will also need a computer etc. Hope this helps.
 
-for the first time: `go get github.com/Financial-Times/nativerw` or update: `go get -u github.com/Financial-Times/nativerw`
+For the first time: `go get github.com/Financial-Times/nativerw` or update: `go get -u github.com/Financial-Times/nativerw`
```
Comment (khatton-ft) [inline README.md]: No vendoring :open_mouth: !

File: db/mongorw_test.go [hunk 2]
```diff
@@ -163,3 +164,101 @@ func TestOpenFailsIfInitialisationFailed(t *testing.T) {
...
+	found := false
+
+	for uuid := range ids {
+		if uuid == expectedResource.UUID {
+			found = true
+		}
+	}
+
+	assert.True(t, found)
+	cancel()
```
Comment (khatton-ft) [inline db/mongorw_test.go]: Should `cancel()` be deferred in case there's a panic in the intervening code? (This comment applies in several places).

Comment (peteclark-ft) [inline db/mongorw_test.go]: Yes it can be (unsure why I didn't)

File: circle.yml [hunk 3]
```diff
@@ -6,7 +6,8 @@ dependencies:
   pre:
     - docker pull mongo:3.0.2
     - docker run -d -P mongo:3.0.2
-    - go get github.com/axw/gocov/gocov; go get github.com/matm/gocov-html; go get -u github.com/jstemmer/go-junit-report
+    - go get github.com/axw/gocov/gocov; go get github.com/matm/gocov-html; go get -u github.com/jstemmer/go-junit-report; go get -u github.com/kardianos/govendor
+    - govendor sync
```
Comment (khatton-ft) [inline circle.yml]: You'll want to exclude vendored files from the coverage report.

**Pull request merged**